includes/Core/Modules/Module.php

```diff
@@ -587,6 +588,25 @@ protected function parse_date_range( $range, $multiplier = 1, $offset = 1, $prev
 		$start_date_offset = $offset + $number_of_days - 1;
```

Comment:
This hasn't changed, but looks odd as `$offset` here already includes `$number_of_days` above for a previous period (from L583). That would mean `$start_date_offset` could be `($last_x_days * $multiplier) + ($last_x_days * $multiplier) + $offset` - is that correct?

Reply:
This is correct, the math here is moving from the end of the period to the beginning of the period, so we need to add number_of_days (including the potential multiplier) again.

The logic works like this:
Say we are requesting a **previous** period, with an offset of 1 (default, ends yesterday) and a period of 7 days.
$offset is 1 + 7 = 8 - the period ends here - 8 days ago
the start of the period is then calculated:
$start_date_offset = $offset + $number_of_days - 1;
= 8 + 7 - 1 = 14 - the period starts 14 days ago. 14days ago - 8 days ago is the 7 day period that is previous to the last 7 days ending yesterday.